php-checks/src/main/java/org/sonar/php/checks/CountInsteadOfEmptyCheck.java

```diff
@@ -89,6 +92,99 @@ private boolean isCountFunction(FunctionCallTree tree) {
+
+  // Checks if empty() is used on the same variable in the surrounding expression context
+  // Returns true if empty() call is found, indicating we should skip raising an issue
+  private boolean isEmptyUsedInExpression(FunctionCallTree countCallTree, @Nullable ExpressionTree countArgument) {
+    // Early return if count() has no argument
+    if (countArgument == null) {
+      return false;
+    }
+
+    Tree current = countCallTree.getParent();
+    while (current != null) {
+      // Stop at statement boundaries or function definitions
+      if (current.is(USED_STATEMENTS)) {
+        break;
+      }
+
+      if ((current.is(AND_OR_TREE) && containsEmptyCallOnVariable((BinaryExpressionTree) current, countArgument))
+        || (current.is(Tree.Kind.CONDITIONAL_EXPRESSION) && containsEmptyCallInTree(current, countArgument))) {
+        return true;
+      }
+
+      current = current.getParent();
+    }
+
+    return false;
+  }
+
+  // Checks both left and right operands of a binary logical expression for empty() calls
+  private boolean containsEmptyCallOnVariable(BinaryExpressionTree logicalExpression, ExpressionTree targetVariable) {
+    return containsEmptyCallInTree(logicalExpression.leftOperand(), targetVariable) ||
+      containsEmptyCallInTree(logicalExpression.rightOperand(), targetVariable);
+  }
+
+  // Recursively searches a tree for empty() calls on the target variable
+  private boolean containsEmptyCallInTree(Tree tree, ExpressionTree targetVariable) {
+    // Check if this node is a function call to empty() with the same variable
+    if (tree.is(Tree.Kind.FUNCTION_CALL) && tree instanceof FunctionCallTree functionCall && isEmptyFunction(functionCall)) {
+      ExpressionTree emptyArgument = CheckUtils.argument(functionCall, "var", 0)
+        .map(CallArgumentTree::value)
+        .orElse(null);
+      if (emptyArgument != null && areSameVariable(emptyArgument, targetVariable)) {
+        return true;
+      }
+    }
+
+    // Recursively check children for nested logical expressions
+    if (tree.is(AND_OR_TREE)) {
+      BinaryExpressionTree binaryTree = (BinaryExpressionTree) tree;
+      return containsEmptyCallInTree(binaryTree.leftOperand(), targetVariable) ||
+        containsEmptyCallInTree(binaryTree.rightOperand(), targetVariable);
+    }
+
+    // Check inside negation operators (!)
+    if (tree.is(Tree.Kind.LOGICAL_COMPLEMENT)) {
+      return containsEmptyCallInTree(((UnaryExpressionTree) tree).expression(), targetVariable);
+    }
+
+    // Check inside parentheses
+    if (tree.is(Tree.Kind.PARENTHESISED_EXPRESSION)) {
+      return containsEmptyCallInTree(((ParenthesisedExpressionTree) tree).expression(), targetVariable);
+    }
+
+    return false;
+  }
+
+  // Checks if a function call tree is a call to the empty() function
+  private boolean isEmptyFunction(FunctionCallTree tree) {
+    return new FunctionCall("empty").test(TreeValues.of(tree, context().symbolTable()));
```

Comment:
We should use a constant here. WDYT? 🙂 
```suggestion
    return EMPTY_FUNCTION.test(TreeValues.of(tree, context().symbolTable()));
```